### PR TITLE
Subgroups - refactor code subgroup functions registration for better …

### PR DESCRIPTION
**Pull request by gwawiork**
…maintainability

This work is related to subgroups next test code and comment in one of the previous review from @kpet https://github.com/KhronosGroup/OpenCL-CTS/pull/533#pullrequestreview-338602913

### CODE REVIEW

File: test_conformance/subgroups/test_workgroup.cpp [hunk 1]
```diff
@@ -621,234 +621,85 @@ template <typename Ty> struct BC
     }
 };
 
+#define G 2000
+#define L 200
+class run_for_type {
```
Comment (kpet) [inline test_conformance/subgroups/test_workgroup.cpp:626]: With `struct` you could avoid having to explicitly add `public`.

File: test_conformance/subgroups/test_workgroup.cpp [hunk 2]
```diff
@@ -621,234 +621,85 @@ template <typename Ty> struct BC
...
+            device_, context_, queue_, num_elements_, "test_scexmin",
+            scexmin_source, 0, useCoreSubgroups_);
+        return error;
+    }
+
+    cl_device_id device_;
+    cl_context context_;
+    cl_command_queue queue_;
+    int num_elements_;
+    bool useCoreSubgroups_;
```
Comment (kpet) [inline test_conformance/subgroups/test_workgroup.cpp:679]: Don't you want the above to be private?

File: test_conformance/subgroups/test_workgroup.cpp [hunk 3]
```diff
@@ -621,234 +621,85 @@ template <typename Ty> struct BC
...
                                          "test_all", all_source, 0,
                                          useCoreSubgroups);
+    run_for_type rft(device, context, queue, num_elements, useCoreSubgroups);
+    error |= rft.run<cl_uint>();
+    error |= rft.run<cl_int>();
+    error |= rft.run<cl_ulong>();
+    error |= rft.run<cl_long>();
+    error |= rft.run<float>();
+    error |= rft.run<double>();
+    // error |= rft.run<cl_half>();
```
Comment (kpet) [inline test_conformance/subgroups/test_workgroup.cpp:701]: We should create an issue to look into this. Why was half disabled?

Comment (gwawiork) [reply test_conformance/subgroups/test_workgroup.cpp:701]: Regarding half type - I do not know why this is commented (I just leave this as is for today) However I have implemented/fixed this in code related with subgroups next.

File: test_conformance/subgroups/test_workgroup.cpp (hunk 2 above)
Comment (gwawiork) [reply test_conformance/subgroups/test_workgroup.cpp:680]: fixed

File: test_conformance/subgroups/test_workgroup.cpp (hunk 3 above)
Comment (kpet) [reply test_conformance/subgroups/test_workgroup.cpp:702]: I think it's worth creating an issue to make sure we don't forget about this.

Comment (gwawiork) [reply test_conformance/subgroups/test_workgroup.cpp:702]: Done #812

**Pull request merged**